text/0000-api-jwt-authentication.md

```diff
@@ -0,0 +1,322 @@
+---
+layout: default
+title: JWT Authentication for Fabric APIs
+nav_order: 3
+---
+
+- Feature Name: api_jwt_authentication
+- Start Date: 2021-01-11
+- RFC PR:
+- Fabric Component: orderer, peer
+- Fabric Issue:
+
+# Summary
+
+This document proposes an OAuth2 compatible authentication mechanism for MSP
+identities that produces authorization tokens suitable for use with APIs
+exposed by Fabric.
+
+# Motivation
+
+With the introduction of the channel participation API, the Fabric orderer
+exposes administrative operations that are authorized based on mutual TLS
+authentication. This approach was chosen as it protects the operations while
+maintaining conventional REST interaction patterns without explicit message
+hashing and signing.
+
+There are two minor issues with this approach:
+
+1. The client CA certificate used to control access to the channel
+   participation API is disconnected from the orderer's local MSP.
+2. It's often difficult for web applications to connect to servers with TLS
+   mutual authentication.
+
+This goal of this proposal is to describe an authentication flow based on
+[RFC7523][RFC7523] that can be used to obtain an access token for the
+channel participation API from an MSP identity.
+
+# Guide-level Explanation
+
+Fabric supports an authentication mechanism based on OAuth 2.0 and OpenID
```

Comment:
This makes it sound like Fabric already supports it.

Reply:
That's because the template says:

> Explain the proposal as if it was already included in Fabric and you were teaching it to another Fabric developer.